packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -223,17 +241,19 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
   }, [isOpen]);
 
   // Update filtered options to the latest value options according to input value
-  const [filteredOptions, setFilteredOptions] = React.useState(options);
+  const [filteredOptions, setFilteredOptions] = React.useState(
+    optionsProp.current,
+  );
   React.useEffect(() => {
     if (inputValue) {
       setFilteredOptions(
-        filterFunction?.(options, inputValue) ??
+        filterFunction?.(optionsProp.current, inputValue) ??
```

Comment:
I see `options` are still in the deps array, so is this change needed for this `useEffecct`?